tracer/test/Datadog.Trace.Tools.dd_dotnet.ArtifactTests/CreatedumpTests.cs

```diff
@@ -313,6 +315,36 @@ public async Task CheckThreadName()
         File.Exists(reportFile.Path).Should().BeTrue();
     }
 
+    [SkippableFact]
+    public async Task SendReportThroughTelemetry()
+    {
+        SkipOn.Platform(SkipOn.PlatformValue.MacOs);
+
+        using var agent = new MockTelemetryAgent(TcpPortProvider.GetOpenPort()) { OptionalHeaders = true };
+
+        using var helper = await StartConsoleWithArgs(
+                               "crash-datadog",
+                               false,
+                               [LdPreloadConfig, ("DD_TRACE_AGENT_URL", $"http://localhost:{agent.Port}")]);
+
+        await helper.Task;
+
+        using var assertionScope = new AssertionScope();
+        assertionScope.AddReportable("stdout", helper.StandardOutput);
+        assertionScope.AddReportable("stderr", helper.ErrorOutput);
+
+        helper.StandardOutput.Should().Contain(CrashReportExpectedOutput);
+
+        var data = agent.WaitForLatestTelemetry(d => d.IsRequestType(TelemetryRequestTypes.RedactedErrorLogs));
+        data.Should().NotBeNull();
+
+        var log = (LogsPayload)data.Payload;
```

Comment:
Just FYI, this is a bit neater in general

data.TryGetPayload<AppStartedPayload>(TelemetryRequestTypes.RedactedErrorLogs)